Jantimizer/QueryOptimiser/QueryOptimiserEquiDepth.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using QueryOptimiser.QueryGenerators;
+using QueryOptimiser.Cost.CostCalculators;
+using QueryParser;
+using QueryParser.Models;
+using Histograms;
+using Histograms.Managers;
+using DatabaseConnector;
+using QueryOptimiser.Cost.Nodes;
+
+namespace QueryOptimiser
+{
+    public class QueryOptimiser : IQueryOptimiser<IHistogram, IDbConnector>
+    {
+        public IParserManager ParserManager { get; set; }
+        public IQueryGenerator QueryGenerator { get; set; }
+        public IHistogramManager<IHistogram, IDbConnector> HistogramManager { get; set; }
+        public ICostCalculator<IHistogram, IDbConnector> CostCalculator { get; set; }
+
+        public QueryOptimiser(IParserManager parserManager, IQueryGenerator queryGenerator, IHistogramManager<IHistogram, IDbConnector> histogramManager)
+        {
+            ParserManager = parserManager;
+            QueryGenerator = queryGenerator;
+            HistogramManager = histogramManager;
```

Comment:
This only works with equi depth histogram, right? So why give the option to give any type of histogram with the manager?